File: react-native-harmony/harmony/cpp/RNOHCorePackage/TurboModules/I18nManagerTurboModule.cpp
#include "RNOHCorePackage/TurboModules/I18nManagerTurboModule.h"

using namespace rnoh;
using namespace facebook;

static jsi::Value __hostFunction_I18nManagerTurboModule_getConstants(
    jsi::Runtime &rt,
    react::TurboModule &turboModule,
    const jsi::Value *args,
    size_t count) {
    return static_cast<ArkTSTurboModule &>(turboModule).call(rt, "getConstants", args, count);
}

I18nManagerTurboModule::I18nManagerTurboModule(const ArkTSTurboModule::Context ctx, const std::string name)
    : ArkTSTurboModule(ctx, name) {
    methodMap_["getConstants"] = MethodMetadata{0, __hostFunction_I18nManagerTurboModule_getConstants};
}

File: tester/harmony/rnoh/src/main/ets/RNOHCorePackage/turboModules/I18nManagerTurboModule.ts
import I18n from '@ohos.i18n';
import { TurboModule } from "../../RNOH/TurboModule";

export class I18nManagerTurboModule extends TurboModule {
  public static readonly NAME = 'I18nManager';

  getConstants() {
    return {
      isRTL: I18n.isRTL(this.ctx.uiAbilityContext.config.language),
      doLeftAndRightSwapInRTL: true
    };
  }
}
